sycl/source/detail/graph_impl.hpp

```diff
@@ -72,6 +72,46 @@ class node_impl {
             std::unique_ptr<sycl::detail::CG> &&CommandGroup)
       : MCGType(CGType), MCommandGroup(std::move(CommandGroup)) {}
 
+  /// Tests if two nodes have the same content,
+  /// i.e. same command group
+  /// @param Node node to compare with
+  bool operator==(const node_impl &Node) {
```

Comment:
We need to be careful here, because the extension specifies [`node`](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc#75-node) to have common reference semantics. I don't see them defining this operator (which might be something that was missed) but we need to make sure they won't define them in terms of this comparison operation, as it wouldn't adhere to the common reference semantics requirements.

Maybe we should add a comment here to avoid it? Unless it is an incoming change that doesn't try to use this, in which case I'm fine without.

Reply:
Added a comment to this function, that this equality operator is used for the internal implementation class only, and shouldn't be used for the public facing class to preserve CRS